Triton_Inference_Server_Python_API/examples/kafka-io/README.md

```diff
@@ -0,0 +1,145 @@
+# (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+# OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+-->
+
+# Triton Inference Server Kafka I/O Deployment
+
+Using the Triton Inference Server In-Process Python API you can
+integrate triton server based models into any Python framework
+to consume the messages from a Kafka topic and produce the inference results
+back to the kafka topic of choice.
+
+This directory contains an example Triton Inference Server
+deployment based on Kafka I/O that uses threads for each of server, consumer and producer.
+
+| [Installation](#installation) | [Run Deployment](#starting-the-pipeline) | [Send Requests](#send-requests-to-deployment) |
+
+
+## Installation
+
+In this Kafka I/O pipeline we deploy a pre-processing stage of tokenization based on `transformers` tokenization module and can be extended to any type of models as needed.
+
+
+### Clone Repository
+```bash
+git clone https://github.com/triton-inference-server/tutorials.git
+cd tutorials/Triton_Inference_Server_Python_API/examples/kafka-io
+```
+
+### Install dependencies
+
+Please note the following command will take many minutes depending on
+your hardware configuration and network connection.
+
+
+```bash
+pip install -r requirements.txt
+```
+
+If triton server is not already installed, install the dependency by using the following command.
+
+```bash
+pip install /opt/tritonserver/python/tritonserver-2.44.0-py3-none-any.whl
+```
+
+In order to run kafka server, we need to download and start the kafka service, execute the below script that starts the kafka service in the background by doing the following:
+1. Download kafka
+2. Start Kafka service by starting Zookeeper and Kafka brokers
+3. Create 2 new topics with names `inference-input` as input queue and `inference-output` to store the inference results
+
+```bash
+chmod +x start-kafka.sh
+./start-kafka.sh
+```
+
+In case, you see any issues starting the server, make sure the system has java installed. If there is no java installed, please follow the instructions below to install java.
+
+```bash
+apt-get update
+apt-get upgrade
+apt install openjdk-17-jdk openjdk-17-jre
+```
+
+Once successfully installed, you should see an output similar to the output shown below
+
+```bash
+java --version
+openjdk 17.0.10 2024-01-16
+OpenJDK Runtime Environment (build 17.0.10+7-Ubuntu-122.04.1)
+OpenJDK 64-Bit Server VM (build 17.0.10+7-Ubuntu-122.04.1, mixed mode, sharing)
+```
+
+## Starting the pipeline
+
+### Start the inference pipeline
+
+Start the pipeline in the background using the following command once you have logged into the triton container
+
+```bash
+export CONSUMER_CONFIGS='{"bootstrap.servers": "localhost:9092", "security.protocol": "PLAINTEXT", "group.id": "triton-server-kafka-consumer"}'
+export PRODUCER_CONFIGS='{"bootstrap.servers": "localhost:9092", "security.protocol": "PLAINTEXT"}'
```

Comment:
```suggestion
export PRODUCER_CONFIGS='{"bootstrap.servers": "localhost:9092", "security.protocol": "PLAINTEXT"}'
export KAFKA_CONSUMER_MAX_WORKER_THREADS=1
```
Since you reference it below, it may be nice to include it here for clarity.

Reply:
Updated